command/loadbot/execution.go

```diff
@@ -0,0 +1,219 @@
+// We create the transactions using createTransactionObjects before the loadbot send them.
+func (c *Configuration) createTransactionObjects() ([]*web3.Transaction, error) {
+	var transactions []*web3.Transaction
+	var nonces = make(map[types.Address]uint64)
+	var numberOfAccounts = uint64(len(c.Accounts))
+
+	// Each loop create one transaction.
+	for i := uint64(0); i < c.TxnToSend; i++ {
+		// Get sender and receiver.
+		from := c.Accounts[i%numberOfAccounts]
+		to := c.Accounts[(i+1)%numberOfAccounts]
+
+		// Get sender nonce.
+		nonce := nonces[from]
+
+		// Create the transaction object.
+		txn := &web3.Transaction{
+			From:     web3.Address(from),
+			To:       (*web3.Address)(&to),
+			Gas:      c.Gas,
+			Value:    c.Value,
+			GasPrice: c.GasPrice.Uint64(),
+			Nonce:    nonce,
+			V:        []byte{1}, // it is necessary to encode in rlp
+		}
+
+		transactions = append(transactions, txn)
+
+		nonces[from] += 1
+	}
+	return transactions, nil
+}
+
+// run is the main method of the loadbot.
+// The TPS is used to determine the rate at which every transaction is sent.
+func (c *Configuration) run(clients []*jsonrpc.Client, txns []*web3.Transaction) *Metrics {
+	ticker := time.NewTicker(1 * time.Second / time.Duration(c.TPS))
+	defer ticker.Stop()
+
+	clientID := 0
+	numberOfClients := len(clients)
+
+	transactionID := 0
+	numberOfTransactions := len(txns)
+
+	var wg sync.WaitGroup
+	ctx := context.Background()
+
+	start := time.Now()
+	metrics := Metrics{
+		m:        sync.Mutex{},
+		Total:    0,
+		Failed:   0,
+		Duration: 0,
+	}
+	defer func() {
+		metrics.Duration = time.Since(start)
+	}()
+
+	fmt.Println("Loadbot started !")
+	for {
+		select {
+		case <-ticker.C:
+			client := clients[clientID%numberOfClients]
+
+			wg.Add(1)
+			go func(txn *web3.Transaction) {
+				defer wg.Done()
+				metrics.m.Lock()
+				metrics.Total += 1
+				metrics.m.Unlock()
+				hash, err := client.Eth().SendTransaction(txn)
+
+				if err != nil {
+					metrics.m.Lock()
+					metrics.Failed += 1
+					metrics.m.Unlock()
+				}
+
+				metrics.m.Lock()
+				metrics.TxnHashes = append(metrics.TxnHashes, hash)
+				metrics.m.Unlock()
+
+			}(txns[transactionID])
+
+			transactionID += 1
+			clientID += 1
+
+			if transactionID == numberOfTransactions {
+				wg.Wait()
+				return &metrics
+			}
+
+		case <-ctx.Done():
+			wg.Wait()
+			return &metrics
+		}
+	}
+}
+
+func waitUntilTxPoolEmpty(ctx context.Context, client txpoolOp.TxnPoolOperatorClient) (*txpoolOp.TxnPoolStatusResp, error) {
+	res, err := tests.RetryUntilTimeout(ctx, func() (interface{}, bool) {
+		subCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		res, _ := client.Status(subCtx, &empty.Empty{})
+		if res != nil && res.Length == 0 {
+			return res, false
+		}
+		fmt.Printf("TxPool not empty, %d transactions remaining..\n", res.Length)
+		return nil, true
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	return res.(*txpoolOp.TxnPoolStatusResp), nil
+}
```

Comment:
Don't we already have this helper method somewhere in the testing framework? If not on the develop branch, then on some other branch definitely. Please consider moving it to a more appropriate place -> helper/tests/testing.go

Reply:
Thanks, I have used the E2E method here. Since we use it outside of the E2E scope, I agree with your idea of moving it to a more appropriate place :+1: 

Added in https://github.com/0xPolygon/polygon-sdk/pull/197/commits/c4b96d1419a2d4800e1f3d952d141b9e0138d1e6